plugin/pkg/auth/authorizer/webhook/webhook.go

```diff
@@ -0,0 +1,174 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package webhook implements the authorizer.Authorizer interface using HTTP webhooks.
+package webhook
+
+import (
+	"errors"
+
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/apis/authorization/v1beta1"
+	"k8s.io/kubernetes/pkg/auth/authorizer"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/client/unversioned/clientcmd"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/runtime/serializer/json"
+	"k8s.io/kubernetes/pkg/runtime/serializer/versioning"
+)
+
+var (
+	// The scheme of this plugin.
+	pluginScheme = runtime.NewScheme()
+
+	encodeVersions = []unversioned.GroupVersion{v1beta1.SchemeGroupVersion}
+	decodeVersions = []unversioned.GroupVersion{v1beta1.SchemeGroupVersion}
+)
+
+func init() {
+	v1beta1.AddToScheme(pluginScheme)
```

Comment:
Yeah, I don't think this should be done like this.  @caesarxuchao created the registration/installation/enablement APIs with the intent that enablement is distinct from usage.

Why should this have a separate scheme overall?  The only other spot I know of with a separate scheme is the kubeconfig files and that was only because it pre-dated groups and we didn't want it exposed like pods.